includes/subscriptions/class-wc-payments-subscriptions-migrator.php

```diff
@@ -265,28 +301,211 @@ private function get_wcpay_subscription_status( $wcpay_subscription ) {
+	 *
+	 * @param string $action_id The Action Scheduler action ID.
+	 * @param array  $error     The error data.
 	 */
-	public function log_unexpected_shutdown() {
-		$error = error_get_last();
-
+	public function log_unexpected_shutdown( $action_id, $error = null ) {
 		if ( ! empty( $error['type'] ) && in_array( $error['type'], [ E_ERROR, E_PARSE, E_COMPILE_ERROR, E_USER_ERROR, E_RECOVERABLE_ERROR ], true ) ) {
 			$this->logger->log( sprintf( '---- ERROR: %s in %s on line %s.', $error['message'] ?? 'No message', $error['file'] ?? 'no file found', $error['line'] ?? '0' ) );
 		}
 	}
+
+	/**
+	 * Logs any unexpected failures that occur while processing a scheduled migrate WCPay Subscription action.
+	 *
+	 * @param string    $action_id The Action Scheduler action ID.
+	 * @param Exception $exception The exception thrown during action processing.
+	 */
+	public function log_unexpected_action_failure( $action_id, $exception ) {
+		$this->logger->log( sprintf( '---- ERROR: %s', $exception->getMessage() ) );
+	}
+
+	/**
+	 * Adds a manual migration tool to WooCommerce > Status > Tools.
+	 *
+	 * This tool is only loaded on stores that have:
+	 *  - WC Subscriptions extension activated
+	 *  - Subscriptions with WooPayments feature disabled
+	 *  - Existing WCPay Subscriptions that can be migrated
+	 *
+	 * @param array $tools List of WC debug tools.
+	 *
+	 * @return array List of WC debug tools.
+	 */
+	public function add_manual_migration_tool( $tools ) {
+		if ( WC_Payments_Features::is_wcpay_subscriptions_enabled() || ! class_exists( 'WC_Subscriptions' ) ) {
+			return $tools;
+		}
+
+		// Get number of WCPay Subscriptions that can be migrated.
+		$wcpay_subscriptions_count = count(
+			wcs_get_orders_with_meta_query(
+				[
+					'status'     => 'any',
+					'return'     => 'ids',
+					'type'       => 'shop_subscription',
+					'limit'      => -1,
+					'meta_query' => [ // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+						[
+							'key'     => WC_Payments_Subscription_Service::SUBSCRIPTION_ID_META_KEY,
+							'compare' => 'EXISTS',
+						],
+					],
+				]
+			)
+		);
+
+		if ( $wcpay_subscriptions_count < 1 ) {
+			return $tools;
+		}
+
+		$disabled = as_next_scheduled_action( $this->scheduled_hook );
+
+		$tools['migrate_wcpay_subscriptions'] = [
+			'name'             => __( 'Migrate Stripe Billing subscriptions', 'woocommerce-payments' ),
+			'button'           => $disabled ? __( 'Migration in progress', 'woocommerce-payments' ) . '&#8230;' : __( 'Migrate Subscriptions', 'woocommerce-payments' ),
+			'desc'             => sprintf(
+				// translators: %1$s and %2$s are <a> tags, %3$d is the number of subscriptions.
+				__( 'This tool will migrate all Stripe Billing subscriptions to tokenized subscriptions with WooPayments. %1$sRead more.%2$sNumber of Stripe Billing subscriptions found: %3$d', 'woocommerce-payments' ),
```

Comment:
Before we forget, we agreed we'd get rid of this read more link until we know where it will point to.